test/api_spec.js

```diff
@@ -33,26 +33,32 @@ describe("API Tests", function () {
     it("Basic proxy constructor", function () {
         expect(proxy).toBeDefined();
         expect(proxy.default_target).toBe(undefined);
-        expect(proxy.target_for_url('/')).toEqual({
+        expect(proxy.target_for_req({url: '/'})).toEqual(
+          jasmine.objectContaining({
```

Comment:
I'd rather keep exact-match tests here, rather than containing tests, if we can. If these are the only keys we should have, but host is added as undefined, then the tests was rightly failing and host probably shouldn't be added, rather than adding host and making the tests insensitive to change.